frontend/src/constants/global.ts

```diff
@@ -3,3 +3,26 @@ export const FULLTIME_HOURS = 37.5;
 
 // A placeholder for authentication token
 export const getAuthenticationToken = (): string => 'xx';
+
+enum DateStrBrand {}
+
+export type DateStr = string & DateStrBrand;
+
+function checkValidDateStr(str: string): str is DateStr {
+  return str.match(/^\d{4}-\d{2}-\d{2}$/) !== null;
+}
+
+export function toDateStr(date: Date | string): DateStr {
+  if (typeof date === 'string') {
+    if (checkValidDateStr(date)) {
+      return date;
+    }
+    throw new Error(`Invalid date string: ${date}`);
+  } else {
+    const dateString = date.toISOString().split('T')[0];
+    if (checkValidDateStr(dateString)) {
+      return dateString;
+    }
+  }
+  throw new Error(`Shouldn't get here (invalid toDateStr provided): ${date}`);
+}
```

Comment:
Date and time libraries can handle this for you.
I imported `date-fns` in another ticket, so we can reuse it here.

Reply:
Does date-fns allow us to do the typing of dates? Or do should I just replace the conversion/ check functions with it?